src/SYCL-Graph/diamondDependency.cpp

```diff
@@ -0,0 +1,103 @@
+/***************************************************************************
+ *
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ **************************************************************************/
+
+#include <sycl/sycl.hpp>
+
+namespace sycl_ext = sycl::ext::oneapi::experimental;
+using namespace sycl;
+
+int main() {
+  constexpr size_t Size = 1024;
+
+  queue Queue{};
```

Comment:
We should check for the `ext_oneapi_limited_graph` aspect and exit early if the device doesn't support it, same for `dotProduct.cpp`. The other tests which use update will require a `ext_oneapi_graph` check.